### PR TITLE
Use regexp in MIME-types based maps

### LINKED ISSUE #220: Need to use regexp in cache_expiration.conf
cache_expiration.conf uses `$sent_http_content_type` to figure out the mimetype.  However, it can also include other things (e.g. `text/html;charset=UTF-8`) which causes the string match to fail.

It needs to use regular expressions to properly figure out the mime-type.

### PR DESCRIPTION
**Pull request by markwoon**
Switched to using regular expressions.

Since we're already using regular expressions, I also simplified the media file and web font mapping.

Fix #220

### CODE REVIEW

File: h5bp/web_performance/cache_expiration.conf [hunk 1]
```diff
@@ -13,76 +13,62 @@
...
-  image/jpeg                            1M;
-  image/png                             1M;
-  image/svg+xml                         1M;
-  image/webp                            1M;
-  video/mp4                             1M;
-  video/ogg                             1M;
-  video/webm                            1M;
+  ~*audio/.*                                 1M;
+  ~*image/.*                                 1M;
+  ~*video/.*                                 1M;
```
[the remark this answers is not in the document]
Comment (LeoColomb) [reply h5bp/web_performance/cache_expiration.conf]: Hmm, but are we sure that `~*image/x-icon\;?` for example is not overridden?

**Comment by markwoon** [comment]
[MIME RFC states](https://www.ietf.org/rfc/rfc2045.txt) that for Content-Type both type and subtype are case insensitive: 

> content := "Content-Type" ":" type "/" subtype
> *(";" parameter)
> ; Matching of media type and subtype
> ; is ALWAYS case-insensitive.

So we need to use `~*` instead of `~`.

As for adding the `\;?`, that can probably be dropped.

File: h5bp/web_performance/cache_expiration.conf (hunk 1 above)
[the remark this answers is not in the document]
Comment (markwoon) [reply h5bp/web_performance/cache_expiration.conf]: I think so, at least according to my reading of the docs:

>If the source value matches more than one of the specified variants, e.g. both a mask and a regular expression match, the first matching variant will be chosen, in the following order of priority:

> 1. string value without a mask
> 1. longest string value with a prefix mask, e.g. “*.example.com”
> 1. longest string value with a suffix mask, e.g. “mail.*”
> 1. first matching regular expression (in order of appearance in a configuration file)
> 1. default value

So it's both the longest and first matching.

**Comment by LeoColomb** [comment]
> Would you prefer the shorter version?

👍 let's keep it as simple as possible! 🙂

**Comment by markwoon** [comment]
Done.

**Pull request merged**